src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,683 @@
+        Set<String> result = new HashSet<String>();
+        String configDropins = serverDirectory.getAbsolutePath() + "/configDropins/" + folderName;
+        File configDropinsFolder = new File(configDropins);
+        String[] configDropinsFileList = configDropinsFolder.list();
+        if (configDropinsFileList == null) {
+            return result;
+        }
+        List<String> configDropinsXmls = new ArrayList<String>();
+        for (String xml : configDropinsFileList){
+            if (xml.endsWith(".xml")) {
+                configDropinsXmls.add(configDropins + "/" +xml);
+            }
+        }
+        if (configDropinsXmls.isEmpty()) {
+            return result;
+        }
+        for (String filename : configDropinsXmls) {
+            result.addAll(getServerXmlFeatures(new File(filename), null));
+        }
+        return result;
+    }
+
+    private static Set<String> getServerXmlFeatures(File serverFile, List<File> parsedXmls) {
+        Set<String> result = new HashSet<String>();
+        List<File> updatedParsedXmls = new ArrayList<File>();
+        updatedParsedXmls.add(serverFile);
+        if (parsedXmls != null){
+            updatedParsedXmls.addAll(parsedXmls);
+        }
+        if (serverFile.exists()) {
+            try {
+                Document doc = new XmlDocument() {
+                    public Document getDocument(File file) throws IOException, ParserConfigurationException, SAXException {
+                        createDocument(file);
+                        return doc;
+                    }
+                }.getDocument(serverFile);
+                Element root = doc.getDocumentElement();
+                NodeList nodes = root.getChildNodes();
+                for (int i = 0; i < nodes.getLength(); i++) {
+                    if (nodes.item(i) instanceof Element) {
+                        Element child = (Element) nodes.item(i);
+                        if ("featureManager".equals(child.getNodeName())) {
+                            result.addAll(parseFeatureManagerNode(child));
+                        } else if ("include".equals(child.getNodeName())){
+                            result = parseIncludeNode(result, serverFile, child, updatedParsedXmls);
+                        }
+                    }
+                }
+            } catch (IOException | ParserConfigurationException | SAXException e) {
+                // just skip this server.xml if it cannot be parsed
+            }
+        }
+        return result;
+    }
+    
+    private static Set<String> parseIncludeNode(Set<String> origResult, File serverFile, Element node,
+            List<File> updatedParsedXmls) {
+        Set<String> result = origResult;
+        String includeFileName = node.getAttribute("location");
+        File includeFile = new File(includeFileName);
+        if (!includeFile.isAbsolute()){
+            includeFile = new File(serverFile.getParentFile().getAbsolutePath(), includeFileName);
+        }
```

Comment:
Else, `includeFile` is absolute, so you will also want the canonical file. So add an else block setting

```
includeFile = includeFile.getCanonicalFile();
```

Reply:
Done.